v2/dtos/requests/event.go

```diff
@@ -0,0 +1,79 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package requests
+
+import (
+	"encoding/json"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+)
+
+// AddEventRequest defines the Request Content for POST event DTO.
+// This object and its properties correspond to the AddEventRequest object in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-data/2.x#/AddEventRequest
+type AddEventRequest struct {
+	common.BaseRequest `json:",inline"`
+	Device             string             `json:"device" validate:"required"`
+	Origin             int64              `json:"origin" validate:"gte=1"`
+	Readings           []dtos.BaseReading `json:"readings"`
+}
+
+// UpdateEventPushedByChecksumRequest defines the Request Content for PUT event as pushed DTO.
+// This object and its properties correspond to the UpdateEventPushedByChecksumRequest object in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-data/2.x#/UpdateEventPushedByChecksumRequest
+type UpdateEventPushedByChecksumRequest struct {
+	common.BaseRequest `json:",inline"`
+	Checksum           string `json:"checksum"`
+}
+
+// Validate satisfies the Validator interface
+func (a AddEventRequest) Validate() error {
+	v2.NewValidator()
+	err := v2.Validate(a)
+	return err
+}
+
+func (a *AddEventRequest) UnmarshalJSON(b []byte) error {
+	var addEvent struct {
+		common.BaseRequest
+		Device   string
+		Origin   int64
+		Readings []dtos.BaseReading
+	}
+	if err := json.Unmarshal(b, &addEvent); err != nil {
+		return v2.NewErrContractInvalid("Failed to unmarshal request body as JSON.")
+	}
+
+	a.RequestID = addEvent.RequestID
+	a.Device = addEvent.Device
+	a.Origin = addEvent.Origin
+	a.Readings = addEvent.Readings
+
+	// validate AddEventRequest DTO
+	if err := a.Validate(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Internal: AddEventReqToEventModels transforms the AddEventRequest DTO array to the Event model array
```

Comment:
This function is Public, not internal, as defined. If it should be internal then change name to `addEventReqToEventModels` or remove `Internal:` from the description.